tests/func/test_diff.py

```diff
@@ -192,3 +192,54 @@ def setUp(self):
     def test(self):
         with self.assertRaises(FileNotInCommitError):
             self.dvc.diff(self.a_ref, target=self.unknown_file)
+
+
+class TestDiffModifiedFile(TestDiff):
+    maxDiff = None
+
+    def setUp(self):
+        super(TestDiffModifiedFile, self).setUp()
+
+        self.new_file_content = "new_test_file_bigger_content_123456789"
+        self.diff_len = len(self.new_file) + len(self.new_file_content)
+        self.create(self.new_file, self.new_file_content)
+        self.dvc.add(self.new_file)
+        self.git.index.add([self.new_file + ".dvc"])
+        self.git.index.commit("change new_file content to be bigger")
+        self.new_checksum = _get_checksum(self.dvc, self.new_file)
+        self.b_ref = self.git.git.rev_parse(self.git.head.commit, short=True)
+
+    def test(self):
+        result = self.dvc.diff(
+            self.a_ref, b_ref=self.b_ref, target=self.new_file
+        )
+        test_dct = {
+            diff.DIFF_A_REF: self.git.git.rev_parse(self.a_ref, short=True),
+            diff.DIFF_B_REF: self.git.git.rev_parse(self.b_ref, short=True),
+            diff.DIFF_LIST: [
+                {
+                    diff.DIFF_NEW_CHECKSUM: self.new_checksum,
+                    diff.DIFF_NEW_FILE: self.new_file,
+                    diff.DIFF_TARGET: self.new_file,
+                    diff.DIFF_SIZE: self.diff_len,
+                }
+            ],
+        }
+        self.assertEqual(test_dct, result)
+
+
+class TestDiffModifiedFileMessage(TestDiffModifiedFile):
+    def test(self):
+        test_dct = {
+            diff.DIFF_IDENT: 0,
+            diff.DIFF_CHANGE: 1,
+            diff.DIFF_NEW: 0,
+            diff.DIFF_DEL: 0,
+            diff.DIFF_SIZE: self.diff_len,
+        }
+
+        msg = CmdDiff._get_dir_changes(test_dct)
```

Comment:
this is not a good idea to test an internal method. Run the tool from CLI or use a public method and mock logger. See some examples around how do we test outputs.

Reply:
Fixed. It now calls "diff" on a directory, previously it called it on a file.

"diff.DIFF_IDENT" returns 2. Shouldn't it be 1? Directory - not changed, file - changed.

Also, in the same  test file there's already a test that calls a private method -> https://github.com/iterative/dvc/blob/master/tests/func/test_diff.py#L75 Should I fix it?

Reply:
> "diff.DIFF_IDENT" returns 2

looks like a bug to me. Should be 0 if directory contains a single file. As far as I understand it's a number of files, not directories.

> Also, in the same test file there's already a test that calls a private method -> https://github.com/iterative/dvc/blob/master/tests/func/test_diff.py#L75 Should I fix it?

yes, that would be great.

Reply:
I plan to  fix that today or tomorrow.

Reply:
Done